test/tests/functional/pbs_hook_exechost_periodic.py

```diff
@@ -74,3 +74,130 @@ def test_multiple_exechost_periodic_hooks(self):
+                      'alarm': '5'}
+        hook_script = """time.sleep(10)"""
+        hook_body = """import time \n"""
+        hook_body += common_periodic_hook_script + hook_script
+        self.server.create_import_hook(hook_name, hook_attrs, hook_body)
+        log_msg = "alarm call while running exechost_periodic hook"
+        log_msg += " '%s', request rejected" % hook_name
+        exp_msg = ["In exechost_periodic hook",
+                   log_msg,
+                   "Non-zero exit status 253 encountered for periodic hook",
+                   "exechost_periodic request rejected by '%s'" % hook_name]
+        for mom in self.moms.values():
+            for msg in exp_msg:
+                mom.log_match(msg)
+
+    @requirements(num_moms=2)
+    def test_exechost_periodic_error(self):
+        """
+        Test exechost_periodic with an unhandled exception in the hook script
+        """
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True'}
+        hook_script = "x"
+        hook_body = common_periodic_hook_script + hook_script
+        self.server.create_import_hook(hook_name, hook_attrs, hook_body)
+
+        common_msg = "PBS server internal error (15011) in "
+        common_msg += "Error evaluating Python script"
+        exp_msg = ["In exechost_periodic hook",
+                   common_msg + ", <class 'NameError'>",
+                   common_msg + ", name 'x' is not defined",
+                   "Non-zero exit status 254 encountered for periodic hook",
+                   "exechost_periodic request rejected by '%s'" % hook_name]
+        for mom in self.moms.values():
+            for msg in exp_msg:
+                mom.log_match(msg)
+
+    @requirements(num_moms=2)
+    def test_exechost_periodic_custom_resc(self):
+        """
+        Test setting custom resource setting on vnode using exechost_periodic
+        hook
+        """
+        self.momB = self.moms.values()[1]
+        self.hostB = self.momB.shortname
+        hook_name = "periodic"
+        hook_attrs = {'event': 'exechost_periodic', 'enabled': 'True'}
+        hook_script = """vn[remote_node].resources_available["foo"] = True"""
+        hook_body = common_periodic_hook_script + hook_script
+        self.server.create_import_hook(hook_name, hook_attrs, hook_body)
+        node_attribs = {'resources_available.foo': True}
+        self.server.expect(NODE, node_attribs, id=self.hostB)
+
+common_periodic_hook_script = """import pbs
+pbs.logmsg(pbs.LOG_DEBUG, "In exechost_periodic hook")
+server_node = pbs.server().name
+pbs.logmsg(pbs.LOG_DEBUG, "server name is %s" % server_node)
+
+vn = pbs.event().vnode_list
+vnodes = pbs.server().vnodes()
+for node in vnodes:
+    if node.name != server_node:
+        remote_node = node.name
+        pbs.logmsg(pbs.LOG_DEBUG, "remote node is %s" % node.name)
+vn = pbs.event().vnode_list
+if remote_node not in vn.keys():
```

Comment:
No need to say vn.keys().  saying 'if remote_node not vn' is the same.  It looks at the keys. (here and below)